docs/content/config/advanced/kubernetes.md

```diff
@@ -4,518 +4,626 @@ title: 'Advanced | Kubernetes'
+      SSL_KEY_PATH: /secrets/ssl/rsa/tls.key
+    ```
 
-    With the configuration shown above, you can **not** dynamically add accounts as the configuration file mounted into the mail server can not be written to.
+    We can also make use of user-provided configuration files, e.g. `user-patches.sh`, `postfix-accounts.cf` and more, to adjust DMS to our likings. We encourage you to have a look at [Kustomize][kustomize] for creating `ConfigMap`s from multiple files, but for now, we will provide a simple, hand-written example. This example is absolutely minimal and only goes to show what can be done.
 
-    Use persistent volumes for production deployments.
+    ```yaml
+    ---
+    apiVersion: v1
+    kind: ConfigMap
 
-### Persistence
+    metadata:
+      name: mailserver.files
 
-Thereafter, we need persistence for our data. Make sure you have a storage provisioner and that you choose the correct `storageClassName`.
+    data:
+      postfix-accounts.cf: |
+        test@example.com|{SHA512-CRYPT}$6$someHashValueHere
+        other@example.com|{SHA512-CRYPT}$6$someOtherHashValueHere
+    ```
 
-```yaml
----
-apiVersion: v1
-kind: PersistentVolumeClaim
+    !!! attention "Static Configuration"
 
-metadata:
-  name: data
+        With the configuration shown above, you can **not** dynamically add accounts as the configuration file mounted into the mail server can not be written to.
 
-spec:
-  storageClassName: local-path
-  accessModes:
-    - ReadWriteOnce
-  resources:
-    requests:
-      storage: 25Gi
-```
+        Use persistent volumes for production deployments.
 
-### Service
+=== "`PersistentVolumeClaim`"
 
-A `Service` is required for getting the traffic to the pod itself. The service is somewhat crucial. Its configuration determines whether the original IP from the sender will be kept. [More about this further down below](#exposing-your-mail-server-to-the-outside-world).
+    Thereafter, we need persistence for our data. Make sure you have a storage provisioner and that you choose the correct `storageClassName`.
 
-The configuration you're seeing does keep the original IP, but you will not be able to scale this way. We have chosen to go this route in this case because we think most Kubernetes users will only want to have one instance.
+    ```yaml
+    ---
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+
+    metadata:
+      name: data
+
+    spec:
+      storageClassName: local-path
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: 25Gi
+    ```
+
+=== "`Service`"
+
+    The `Service`'s configuration determines whether the original IP from the sender will be kept. [More about this further down below](#exposing-your-mail-server-to-the-outside-world). The configuration you're seeing does keep the original IP, but you will not be able to scale this way. We have chosen to go this route in this case because we think most Kubernetes users will only want to have one instance.
```

Comment:
I dropped the first sentence because it's stating the obvious. In the K8s docs page, it's the equivalent of saying Docker is an application for running containers 😆 I found it to be an improvement because I am not a fan of very short paragraphs. I also cannot understand why we suddenly use bullet points for normal sentences - why do it here now? 

I will take a middle-ground now between your suggestion and my approach.

Reply:
> With my lack of experience, the `name` and `targetPort` fields having the same values is a tad unclear to me from this snippet alone 😅

`targetPort` references the port that is targeted when looking at the `Deployment`. `name` is the name of port when looking at the `Service`, which can be references from an `IngressRoute` for example.

Reply:
> The bullet points above are focused on the IP concern and something about how this is preserved from the config but prevents scaling beyond a single instance which from the snippet alone is not obvious why that is not scalable 🤷‍♂️ This IP concern seems more suited to be in an info admonition instead, beneath the config snippet?

Good point; I will adjust the content.